src/backends/policy_engine/souffle/datalog_fact_policy.h

```diff
@@ -0,0 +1,54 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_BACKENDS_POLICY_ENGINE_SOUFFLE_DATALOG_FACT_POLICY_H_
+#define SRC_BACKENDS_POLICY_ENGINE_SOUFFLE_DATALOG_FACT_POLICY_H_
+
+#include <fstream>
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "src/backends/policy_engine/policy.h"
+#include "src/backends/policy_engine/souffle/utils.h"
+
+namespace raksha::backends::policy_engine::souffle {
+
+class DatalogFactPolicy : public Policy {
+ public:
+  DatalogFactPolicy(absl::string_view relation_name,
+                    absl::string_view facts_string)
+      : relation_name_(relation_name), facts_string_(facts_string) {}
+
+  absl::Status OutputAsDlFactsToDirectory(
+      const std::filesystem::path &output_path) const override {
+    return WriteFactsStringToFactsFile(output_path, relation_name_,
+                                       facts_string_);
+  }
+
+ private:
+  std::string relation_name_;
+  std::string facts_string_;
+};
```

Comment:
Haven't thought this through, but I don't know if this is right abstraction of policy for the long run. Let us get rid of this because we don't need this right now?

If you want to keep this class, then make it simple and move the logic in OutputAsDlFactsToDirectory back to souffle_policy_checker?

Reply:
Got rid of this, put the string directly in `Policy` for now.